quickstarts/microsoft.apimanagement/api-management-create-all-resources/main.bicep

```diff
@@ -0,0 +1,266 @@
+  parent: apiManagementServiceName_exampleApiWithPolicy
+  name: 'policy'
+  properties: {
+    value: apiPolicy
+    format: 'xml'
+  }
+}
+
+resource apiManagementServiceName_exampleProduct 'Microsoft.ApiManagement/service/products@2022-08-01' = {
+  parent: apiManagementService
+  name: 'exampleProduct'
+  properties: {
+    displayName: 'Example Product Name'
+    description: 'Description for example product'
+    terms: 'Terms for example product'
+    subscriptionRequired: true
+    approvalRequired: false
+    subscriptionsLimit: 1
+    state: 'published'
+  }
+}
+
+resource apiManagementServiceName_exampleProduct_exampleApi 'Microsoft.ApiManagement/service/products/apis@2022-08-01' = {
+  parent: apiManagementServiceName_exampleProduct
+  name: 'exampleApi'
+  dependsOn: [
+    apiManagementServiceName_exampleApi
+  ]
+}
+
+resource apiManagementServiceName_exampleProduct_policy 'Microsoft.ApiManagement/service/products/policies@2022-08-01' = {
+  parent: apiManagementServiceName_exampleProduct
+  name: 'policy'
+  properties: {
+    value: productPolicy
+    format: 'xml'
+  }
+}
+
+resource apiManagementServiceName_exampleUser1 'Microsoft.ApiManagement/service/users@2022-08-01' = {
+  parent: apiManagementService
+  name: 'exampleUser1'
+  properties: {
+    firstName: 'ExampleFirstName1'
+    lastName: 'ExampleLastName1'
+    email: 'ExampleFirst1@example.com'
+    state: 'active'
+    note: 'note for example user 1'
+  }
+}
+
+resource apiManagementServiceName_exampleUser2 'Microsoft.ApiManagement/service/users@2022-08-01' = {
+  parent: apiManagementService
+  name: 'exampleUser2'
+  properties: {
+    firstName: 'ExampleFirstName2'
+    lastName: 'ExampleLastName2'
+    email: 'ExampleFirst2@example.com'
+    state: 'active'
+    note: 'note for example user 2'
+  }
+}
+
+resource apiManagementServiceName_exampleUser3 'Microsoft.ApiManagement/service/users@2022-08-01' = {
+  parent: apiManagementService
+  name: 'exampleUser3'
+  properties: {
+    firstName: 'ExampleFirstName3'
+    lastName: 'ExampleLastName3'
+    email: 'ExampleFirst3@example.com'
+    state: 'active'
+    note: 'note for example user 3'
+  }
+}
+
+resource apiManagementServiceName_exampleNamedValue 'Microsoft.ApiManagement/service/namedValues@2022-08-01' = {
+  parent: apiManagementService
+  name: 'exampleNamedValue'
+  properties: {
+    displayName: 'propertyExampleName'
+    value: 'propertyExampleValue'
+    tags: [
+      'exampleTag'
+    ]
+  }
+}
+
+resource apiManagementServiceName_examplesubscription1 'Microsoft.ApiManagement/service/subscriptions@2022-08-01' = {
+  parent: apiManagementService
+  name: 'examplesubscription1'
+  properties: {
+    displayName: 'subscription 1'
+    scope: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/exampleServiceName/products/exampleProduct'
```

Comment:
are these right? It looks like `{subscriptionId}` and `{resourceGroupName}` need to be replaced? Can these hardcoded values be replaced with `apiManagementServiceName_exampleProduct.id` and `apiManagementServiceName_exampleUser1.id`?